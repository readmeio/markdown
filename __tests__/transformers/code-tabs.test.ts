import { mdast, hast } from '../../index';

describe('Code Tabs Transformer', () => {
  it('wraps single code blocks with tabs if they have a lang set', () => {
    const md = `
\`\`\`javascript
const languageSet = true;
\`\`\`
`;

    const tree = mdast(md);
    expect(tree).toMatchSnapshot();
  });

  it('wraps single code blocks with tabs if they have a title set', () => {
    const md = `
\`\`\`javascript Testing
const languageSet = true;
\`\`\`
`;

    const tree = mdast(md);
    expect(tree).toMatchSnapshot();
  });

  it('can parse code tabs', () => {
    const md = `
\`\`\`
First code block
\`\`\`
\`\`\`
Second code block
\`\`\`
`;

    expect(mdast(md)).toMatchSnapshot();
  });

  it('can parse lang and meta', () => {
    const md = `
\`\`\`javascript First Title
First code block
\`\`\`
\`\`\`text
Second code block
\`\`\`
`;
    const ast = mdast(md);

    expect(ast.children[0].children[0]).toStrictEqual(
      expect.objectContaining({ lang: 'javascript', meta: 'First Title' }),
    );
    expect(ast.children[0].children[1]).toStrictEqual(expect.objectContaining({ lang: 'text' }));
  });

  it('allows code tabs within html blocks', () => {
    const md = `
<p>

\`\`\`
First code block
\`\`\`
\`\`\`
Second code block
\`\`\`

</p>
`;

    expect(hast(md)).toMatchSnapshot();
  });
});
