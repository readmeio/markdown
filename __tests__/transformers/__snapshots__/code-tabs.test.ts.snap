// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Code Tabs Transformer > allows code tabs within html blocks 1`] = `
{
  "children": [
    {
      "attributes": [],
      "children": [
        {
          "children": [
            {
              "children": [
                {
                  "children": [
                    {
                      "type": "text",
                      "value": "First code block
",
                    },
                  ],
                  "position": {
                    "end": {
                      "column": 4,
                      "line": 6,
                      "offset": 30,
                    },
                    "start": {
                      "column": 1,
                      "line": 4,
                      "offset": 6,
                    },
                  },
                  "properties": {},
                  "tagName": "code",
                  "type": "element",
                },
              ],
              "position": {
                "end": {
                  "column": 4,
                  "line": 6,
                  "offset": 30,
                },
                "start": {
                  "column": 1,
                  "line": 4,
                  "offset": 6,
                },
              },
              "properties": {},
              "tagName": "pre",
              "type": "element",
            },
            {
              "children": [
                {
                  "children": [
                    {
                      "type": "text",
                      "value": "Second code block
",
                    },
                  ],
                  "position": {
                    "end": {
                      "column": 4,
                      "line": 9,
                      "offset": 56,
                    },
                    "start": {
                      "column": 1,
                      "line": 7,
                      "offset": 31,
                    },
                  },
                  "properties": {},
                  "tagName": "code",
                  "type": "element",
                },
              ],
              "position": {
                "end": {
                  "column": 4,
                  "line": 9,
                  "offset": 56,
                },
                "start": {
                  "column": 1,
                  "line": 7,
                  "offset": 31,
                },
              },
              "properties": {},
              "tagName": "pre",
              "type": "element",
            },
          ],
          "position": {
            "end": {
              "column": 4,
              "line": 9,
              "offset": 56,
            },
            "start": {
              "column": 1,
              "line": 4,
              "offset": 6,
            },
          },
          "properties": {},
          "tagName": "div",
          "type": "element",
        },
      ],
      "name": "p",
      "position": {
        "end": {
          "column": 5,
          "line": 11,
          "offset": 62,
        },
        "start": {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "mdxJsxFlowElement",
    },
  ],
  "position": {
    "end": {
      "column": 1,
      "line": 12,
      "offset": 63,
    },
    "start": {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Code Tabs Transformer > can parse code tabs 1`] = `
{
  "children": [
    {
      "children": [
        {
          "lang": null,
          "meta": null,
          "position": {
            "end": {
              "column": 4,
              "line": 4,
              "offset": 25,
            },
            "start": {
              "column": 1,
              "line": 2,
              "offset": 1,
            },
          },
          "type": "code",
          "value": "First code block",
        },
        {
          "lang": null,
          "meta": null,
          "position": {
            "end": {
              "column": 4,
              "line": 7,
              "offset": 51,
            },
            "start": {
              "column": 1,
              "line": 5,
              "offset": 26,
            },
          },
          "type": "code",
          "value": "Second code block",
        },
      ],
      "position": {
        "end": {
          "column": 4,
          "line": 7,
          "offset": 51,
        },
        "start": {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "code-tabs",
    },
  ],
  "position": {
    "end": {
      "column": 1,
      "line": 8,
      "offset": 52,
    },
    "start": {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Code Tabs Transformer > wraps single code blocks with tabs if they have a lang set 1`] = `
{
  "children": [
    {
      "children": [
        {
          "lang": "javascript",
          "meta": null,
          "position": {
            "end": {
              "column": 4,
              "line": 4,
              "offset": 44,
            },
            "start": {
              "column": 1,
              "line": 2,
              "offset": 1,
            },
          },
          "type": "code",
          "value": "const languageSet = true;",
        },
      ],
      "position": {
        "end": {
          "column": 4,
          "line": 4,
          "offset": 44,
        },
        "start": {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "code-tabs",
    },
  ],
  "position": {
    "end": {
      "column": 1,
      "line": 5,
      "offset": 45,
    },
    "start": {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Code Tabs Transformer > wraps single code blocks with tabs if they have a title set 1`] = `
{
  "children": [
    {
      "children": [
        {
          "lang": "javascript",
          "meta": "Testing",
          "position": {
            "end": {
              "column": 4,
              "line": 4,
              "offset": 52,
            },
            "start": {
              "column": 1,
              "line": 2,
              "offset": 1,
            },
          },
          "type": "code",
          "value": "const languageSet = true;",
        },
      ],
      "position": {
        "end": {
          "column": 4,
          "line": 4,
          "offset": 52,
        },
        "start": {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "code-tabs",
    },
  ],
  "position": {
    "end": {
      "column": 1,
      "line": 5,
      "offset": 53,
    },
    "start": {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
